### PR TITLE
Fix broken unit tests

### PR DESCRIPTION
**Pull request by Sword-Smith**
Addressing #64 
Fixes:
- adds ethereumjs-util to package.json
- Fixes broken tests in `test/sign.js`
- Upgrading to to latest nodejs (v12.16.1) will make the import of the ripple library work
- Fixes a problem in a Stellar unit test when upgrading to node 12
- In my setup, the tests only work when run sequentially since we are not awaiting
  when writing to the database and `close()` is called on the DB connection before
  writing is completed.
  `node.exe ./node_modules/mocha/bin/_mocha --timeout 20000 .\test\app.js`,
  `node.exe ./node_modules/mocha/bin/_mocha --timeout 20000 .\test\sign.js`,
  `node.exe ./node_modules/mocha/bin/_mocha --timeout 20000 .\test\admin.js`
- `npm test` works correctly if a `sleep(500)` is inserted before the call to `testutils.mongoose.connection.close();`. A better solution might be to add `await` on each DB save.

### CODE REVIEW

**Comment by Sword-Smith** [comment]
I made a force push to improve the regex checking if the BIP32 seed is valid.

File: app/sign.js [hunk 1]
```diff
@@ -236,9 +243,20 @@ const signEthTx = function(recoveryRequest, key, skipConfirm, isToken) {
...
   }
 
-  key = promptForConfirmationAndKey(recoveryRequest, outputs, skipConfirm, key);
+  // When generating signatures, we don't currently use EIP155 but this could
+  // be activated if we wanted to. This would protect against replay attacks on other
+  // blockchains, such as Ethereum Classic. To activate the EIP155, we would have to
+  // know the chain ID of the Ethereum blockchains we are using as this value goes
+  // into the V field when using EIP155.
+  // cf. https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+  const useEip155 = false;
```
Comment (cooncesean) [inline app/sign.js]: That is really interesting - thx for adding. 

So, when enabling this, would a user have to flip the flag here? If time and/or scope creep weren't an issue, would it make sense to move this type of configuration somewhere else within the app?

File: app/sign.js [hunk 2]
```diff
@@ -236,9 +243,20 @@ const signEthTx = function(recoveryRequest, key, skipConfirm, isToken) {
...
 
-  transaction.sign(getBackupSigningKey(key, recoveryRequest.backupKey));
+  key = promptForConfirmationAndKey(recoveryRequest, outputs, skipConfirm, key);
+  const signingKey = Buffer.from(getBackupSigningKey(key, recoveryRequest.backupKey), "hex");
+  const signature = EthUtil.ecsign(transaction.hash(useEip155), signingKey, transaction.chainId);
+  transaction.v = signature.v; // Change this if activating EIP155
```
Comment (cooncesean) [inline app/sign.js]: thanks for the inline comment 👍

File: app/sign.js (hunk 1 above)
Comment (Sword-Smith) [reply app/sign.js]: Do you necessarily need to be able to generate pre-EIP155 signatures? If not, I would just lean towards always using EIP155 signatures. I don't see a reason not to.

That means that this boolean value could be removed and `true` being hardcoded into the `transaction.hash` function call.

This file already contains values like `EOS_MAINNET_CHAIN_ID` and other hardcoded parameters for various blockchains so I imagine that the `CHAIN_ID` values for various Ethereum blockchains could also be contained here. You could then use the `coin` field in your test cases to hold information about which Ethereum blockchain you are signing for.

**Comment by Sword-Smith** [comment]
Sorry. I accidently pressed a button to request a review from you @cooncesean. Didn't mean to be that pushy xD

File: app/sign.js (hunk 1 above)
Comment (Sword-Smith) [reply app/sign.js]: If I knew which testnet blockchain `teth` referred to, I could implement EIP155. But I don't know if you are always using the same test net? I couldn't find any references to `etc` (Ethereum Classic) in this repo, so I don't assume this repo supports Ethereum Classic, so we don't have to worry about that blockchain.

I looked for the txids from the ETH tests on all of etherscan's blockexplorers (main net, Ropsten, Rinkeby, ...) in order to figure out which test net they were published on. But I couldn't find them there so I guess they were never published.

Comment (tylerlevine) [reply app/sign.js]: We use kovan: https://github.com/BitGo/BitGoJS/blob/master/modules/statics/src/networks.ts#L176

**Comment by tylerlevine** [comment]
@cooncesean should @Sword-Smith sign his commits?

**Comment by Sword-Smith** [comment]
I force-pushed to add signatures. All commits should be signed by me now. If you want to activate EIP155, do you want me to do that in another pull request? If you want it in this pull request, I will happily add it here.

**Comment by cooncesean** [comment]
> I force-pushed to add signatures. 

👍 

> If you want to activate EIP155, do you want me to do that in another pull request? If you want it in this pull request, I will happily add it here.

Let's keep that out of scope for now. Thanks for this -- I'll let you and Tyler handle logistics around landing this code.

**Comment by Sword-Smith** [comment]
> 
> 
> lgtm. I still can't get the tests passing even with your `sleep()` trick, but let's handle that in a follow up. Thanks @Sword-Smith!

- Which node version are you on? I am running v12.16.1.
- Does it work if you run the tests sequentially with the command
  `node ./node_modules/mocha/bin/_mocha --timeout 20000 test/sign.js && node ./node_modules  /mocha/bin/_mocha --timeout 20000 test/admin.js && node ./node_modules/mocha/bin/_mocha   --timeout 20000 test/app.js`
- The sleep that I added modifies the `after` hook of the mocha test framework like this:
```
  async function sleep(ms) {
    return await new Promise(resolve => setTimeout(resolve, ms));
  }

  after(function() {
    sleep(500).then(() => {
      testutils.mongoose.connection.close();
    });
  });
```
